### PR TITLE
[build] Move a number of ATen checks out of Dependencies.cmake

### PR DESCRIPTION
**Pull request by orionr**
cc @Yangqing @mingzhe09088 @anderspapitto @mingzhe09088

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -289,6 +289,13 @@ if (USE_ASAN)
     set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_STATIC_LINKER_FLAGS_DEBUG} -fsanitize=address")
 endif()
 
+if (MSVC)
```
Comment (Yangqing) [inline CMakeLists.txt]: nit: this should be in cmake/MiscCheck.cmake? We have a much better commented flags setting here: https://github.com/pytorch/pytorch/blob/master/cmake/MiscCheck.cmake#L157

**Comment by Yangqing** [comment]
Thanks! One nit commit and the rest looks good.

File: CMakeLists.txt (hunk 1 above)
Comment (orionr) [reply CMakeLists.txt]: Good point - moving there.

**Comment by orionr** [comment]
Build timeouts on a few, but otherwise looks good.

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**